controllers/humiocluster_pods.go

```diff
@@ -932,28 +926,9 @@ func (r *HumioClusterReconciler) podsMatch(hnp *HumioNodePool, pod corev1.Pod, d
 	return true, nil
 }
 
-func (r *HumioClusterReconciler) getRestartPolicyFromPodInspection(pod, desiredPod corev1.Pod) (string, error) {
-	humioContainerIdx, err := kubernetes.GetContainerIndexByName(pod, humioContainerName)
-	if err != nil {
-		return "", err
-	}
-	desiredHumioContainerIdx, err := kubernetes.GetContainerIndexByName(desiredPod, humioContainerName)
-	if err != nil {
-		return "", err
-	}
-	if pod.Spec.Containers[humioContainerIdx].Image != desiredPod.Spec.Containers[desiredHumioContainerIdx].Image {
-		return PodRestartPolicyRecreate, nil
-	}
-
-	if envVarValue(pod.Spec.Containers[humioContainerIdx].Env, "EXTERNAL_URL") != envVarValue(desiredPod.Spec.Containers[desiredHumioContainerIdx].Env, "EXTERNAL_URL") {
-		return PodRestartPolicyRecreate, nil
-	}
-
-	return PodRestartPolicyRolling, nil
-}
-
-func (r *HumioClusterReconciler) getPodDesiredLifecycleState(hnp *HumioNodePool, foundPodList []corev1.Pod, attachments *podAttachments) (podLifecycleState, error) {
+func (r *HumioClusterReconciler) getPodDesiredLifecycleState(hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool, foundPodList []corev1.Pod, attachments *podAttachments) (podLifecycleState, error) {
```

Comment:
I don't see `hc *humiov1alpha1.HumioCluster` being used anywhere in this method.